src/test/java/io/vertx/core/eventbus/LocalEventBusTest.java

```diff
@@ -1479,5 +1481,113 @@ public void testCloseSender() {
+    DeliveryOptions noTimeout = new DeliveryOptions().setSendTimeout(Long.MAX_VALUE);
+    MessageConsumer<?> consumer = vertx.eventBus().consumer(ADDRESS1);
+    consumer.handler(message -> message.reply(null));
+    consumer.completionHandler(__ -> {
+      consumer.pause();
+      consumer.setMaxBufferedMessages(0);
+      vertx.eventBus().request(ADDRESS1, null, noTimeout, onFailure(err -> {
+        assertTrue(err instanceof ReplyException);
+        assertEquals(ReplyFailure.TIMEOUT, ((ReplyException) err).failureType());
+        testComplete();
+      }));
+    });
+    await();
+  }
+
+  @Test
+  public void testEarlyTimeoutWhenSetMaxBufferedMessages() {
+    DeliveryOptions noTimeout = new DeliveryOptions().setSendTimeout(Long.MAX_VALUE);
+    MessageConsumer<?> consumer = vertx.eventBus().consumer(ADDRESS1);
+    consumer.handler(message -> message.reply(null));
+    consumer.completionHandler(__ -> {
+      consumer.pause();
+      vertx.eventBus().request(ADDRESS1, null, noTimeout, onFailure(err -> {
+        assertTrue(err instanceof ReplyException);
+        assertEquals(ReplyFailure.TIMEOUT, ((ReplyException) err).failureType());
+        testComplete();
+      }));
+      vertx.setTimer(1000, id -> {
+        // Give some time to the event bus to deliver the message to the consumer
+        consumer.setMaxBufferedMessages(0);
+      });
+    });
+    await();
+  }
+
+  @Test
+  public void testEarlyTimeoutOnHandlerUnregistration() {
+    DeliveryOptions noTimeout = new DeliveryOptions().setSendTimeout(Long.MAX_VALUE);
+    MessageConsumer<?> consumer = vertx.eventBus().consumer(ADDRESS1);
+    consumer.handler(message -> message.reply(null));
+    consumer.completionHandler(__ -> {
+      vertx.eventBus().request(ADDRESS1, null, noTimeout, onFailure(err -> {
+        assertTrue(err instanceof ReplyException);
+        assertEquals(ReplyFailure.TIMEOUT, ((ReplyException) err).failureType());
+        testComplete();
+      }));
+      consumer.unregister();
+    });
+    await();
+  }
+
+  @Test
+  public void testEarlyTimeoutOfBufferedMessagesOnHandlerUnregistration() {
+    DeliveryOptions noTimeout = new DeliveryOptions().setSendTimeout(Long.MAX_VALUE);
+    MessageConsumer<?> consumer = vertx.eventBus().consumer(ADDRESS1);
+    consumer.handler(message -> message.reply(null));
+    consumer.completionHandler(__ -> {
+      consumer.pause();
+      vertx.eventBus().request(ADDRESS1, null, noTimeout, onFailure(err -> {
+        assertTrue(err instanceof ReplyException);
+        assertEquals(ReplyFailure.TIMEOUT, ((ReplyException) err).failureType());
+        testComplete();
+      }));
+      vertx.setTimer(1000, id -> {
+        // Give some time to the event bus to deliver the message to the consumer
+        consumer.unregister();
+      });
+    });
+    await();
+  }
+
+  @Test
+  @Repeat(times = 2)
+  public void testEarlyTimeoutWhenClosingDestinationContext() {
```

Comment:
I don't think it is necessary to test this case, as the unregistration already takes care of testing that an unregistered consumer will fail